scylla/src/network/connection_pool.rs

```diff
@@ -61,6 +64,12 @@ pub(crate) struct PoolConfig {
     pub(crate) pool_size: PoolSize,
     pub(crate) can_use_shard_aware_port: bool,
     pub(crate) keepalive_interval: Option<Duration>,
+    #[cfg(feature = "metrics")]
+    // TODO: The metrics should definitely not be stored here,
+    //       but it was the easiest way to pass it to the refiller.
+    //       It could be refactored to be passed as a parameter to the refiller,
+    //       but it would require a lot of changes in the code.
+    pub(crate) metrics: Option<Arc<Metrics>>,
```

Comment:
:wrench: I refactored passing `Metrics` through the code. It's present on my fork; I've opened a PR to your branch.